packages/ethereum/contracts/HoprChannels.sol

```diff
@@ -246,190 +320,75 @@ contract HoprChannels is IERC777Recipient, ERC1820Implementer {
         require(amount == amount1.add(amount2), "amount sent must be equal to amount specified");
 
         //require(from == account1 || from == account2, "funder must be either account1 or account2");
-        _fundChannel(account1, account2, amount1, amount2);
+
+        if (amount1 > 0){
+          _fundChannel(account1, account2, amount1);
+        }
+        if (amount2 > 0){
+          _fundChannel(account2, account1, amount2);
+        }
     }
 
     // internal code
 
     /**
      * @dev Funds a channel, then emits
      * {ChannelUpdate} event.
-     * @param account1 the address of account1
-     * @param account2 the address of account2
-     * @param amount1 amount to fund account1
-     * @param amount2 amount to fund account2
+     * @param source the address of the channel source
+     * @param dest the address of the channel destination
+     * @param amount amount to fund account1
      */
     function _fundChannel(
-        address account1,
-        address account2,
-        uint256 amount1,
-        uint256 amount2
+        address source,
+        address dest,
+        uint256 amount
     ) internal {
-        require(account1 != account2, "accountA and accountB must not be the same");
-        require(account1 != address(0), "accountA must not be empty");
-        require(account2 != address(0), "accountB must not be empty");
-        require(amount1 > 0 || amount2 > 0, "amountA or amountB must be greater than 0");
-
-        address partyA;
-        address partyB;
-        uint256 amountA;
-        uint256 amountB;
-        
-        if (_isPartyA(account1, account2)){
-          partyA = account1;
-          partyB = account2;
-          amountA = amount1;
-          amountB = amount2;
-        } else {
-          partyA = account2;
-          partyB = account1;
-          amountA = amount2;
-          amountB = amount1;
-        }
-        (,,, Channel storage channel) = _getChannel(partyA, partyB);
+        _validateSourceAndDest(source, dest);
+        require(amount > 0, "amount must be greater than 0");
 
-        require(channel.status != ChannelStatus.PENDING_TO_CLOSE, "Cannot fund a closing channel");
-        
+        (, Channel storage channel) = _getChannel(source, dest);
+        require(channel.status != ChannelStatus.PENDING_TO_CLOSE, "Cannot fund a closing channel"); 
         if (channel.status == ChannelStatus.CLOSED) {
           // We are reopening the channel
           channel.channelEpoch = channel.channelEpoch.add(1);
-          channel.status = ChannelStatus.OPEN;
-          channel.partyATicketIndex = 0;
-          channel.partyBTicketIndex = 0;
+          if (channel.commitment != bytes32(0)) {
```

Comment:
this means we can never _reset_ the commitment, which I was leaving open for the future (you want to _revoke_ your commitment to a channel).

Because of this, I think checking for commitment == 0 is better.

Reply:
I don't think so. This makes sure that we can reach the state WAITING_FOR_COMMITMENT only once, namely when the channel goes from CLOSED to sth else for the very first time. For every subsequent call, we call bumpCommitment which immediately bumps the commitment, so not further is required by the counterparty

Reply:
exactly. I don't want to limit it to _once_ - in order to keep the option for revoking commitments as I mentioned.

Reply:
Imo, we should not be able to *revoke* any commitment because this could allow the counterparty to tweak the entropy used to decide whether tickets are a win or not.